projects/systelab-components/src/lib/image-viewer/image-viewer.component.ts

```diff
@@ -0,0 +1,377 @@
+					this.resizeZoomSelectorAndImage();
+				}
+				this.chref.detectChanges();
+				this.imgParams.sliderZoomPct = this.getSliderPct();
+			}
+			this.deactivateZoomSelector();
+			this.imageClicked = false;
+		} else {
+			this.zoomArea.scrollTop = this.viewPort.scrollTop;
+			this.zoomArea.scrollLeft = this.viewPort.scrollLeft;
+		}
+	}
+
+	public getWidth(): string {
+		if (this.zoomEnabled) {
+			return this.imgParams.width + 'px';
+		} else {
+			return this.imgParams.sliderZoomPct ? this.image.naturalWidth * this.imgParams.sliderZoomPct / 100 + 'px'
+				: undefined;
+		}
+	}
+
+	public getFilter(): string {
+		return this.imgParams.filter;
+	}
+
+	public setFilter(filter: string): void {
+		this.imgParams.filter = filter;
+	}
+
+	public doAdjust(): void {
+		this.imgParams.sliderZoomPct = this.getInitialZoom();
+		this.dragEnabled = false;
+		this.zoomEnabled = false;
+	}
+
+	public toggleZoomByArea(): void {
+		if (this.zoomEnabled) {
+			this.zoomEnabled = false;
+			this.dragEnabled = this.imageOverflowViewport();
+		} else {
+			this.scaleImage(1);
+			this.zoomEnabled = true;
+			this.dragEnabled = false;
+		}
+	}
+
+	public sliderZoomChanged(): void {
+		this.dragEnabled = this.imageOverflowViewport();
+		this.zoomEnabled = false;
+		this.scaleImage(1);
+	}
+
+	public setInitialValues(): void {
+		this.viewPort = this.elementRef.nativeElement;
+		this.image = this.imageViewerImg.nativeElement;
+		this.wrapper = this.imageViewerWrapper.nativeElement;
+
+		// Set zoom scale width and marks
+		this.zoomScale.zoomScaleLength = this.viewPort.querySelector('input[type="range"]').offsetWidth;
+		this.zoomScale.zoomScaleChunk = Math.round(this.zoomScale.zoomScaleLength / 6);
+		// Set zoom to fit the image
+		this.imgParams.sliderZoomPct = this.getInitialZoom();
+		// Min zoom 10% smaller than initial zoom
+		this.sliderZoomMin = this.imgParams.sliderZoomPct - 10;
+		// Max zoom to double the real size of the image
+		this.sliderZoomMax = 200;
+		// Calculate the mark matching the real size of the image (x1 scale)
+		this.zoomScale.sliderMarkLeft = (100 - this.sliderZoomMin) / (this.sliderZoomMax - this.sliderZoomMin)
+			* (this.zoomScale.zoomScaleLength - this.zoomScale.sliderThumb) + this.zoomScale.sliderThumb;
+
+	}
+
+	public isFilterEnabled(action: string): boolean {
+		return this.imgParams.filter === action;
+	}
+
+	public getDropMainDownLabel(dropDownLabel: string): string {
+		return dropDownLabel.substring(0, dropDownLabel.indexOf('|'));
+	}
+
+	public getDropDownLabels(dropDownLabel: string): string[] {
+		return dropDownLabel.slice(dropDownLabel.indexOf('|') + 1)
+			.split(';');
+	}
+
+	public getDropDownAction(dropDownActions: string, i: number): string {
+		return dropDownActions.split(';')[i];
+	}
+
+	public getImageFiltersHtml(): SafeHtml {
+		return this.sanitizer.bypassSecurityTrustHtml(this.imageFilters);
+	}
+
+	public getFilterUrl(): string {
+		return this.imgParams.filter ? `url(#${this.imgParams.filter})` : '';
```

Comment:
Why to use url svg colors matrix? It would be easier use simple filters as "contrast(200%), blur(...) ' etc